src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,185 @@
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medicationsRefDoc = medicationsRef.document(dateName);
+        ApiFuture<DocumentSnapshot> future = medicationsRefDoc.get();
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC, MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.toObject(MedicationEntity.class);
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationData(String owner, String petName) {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        String aux;
+            ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments = null;
+        try {
+            medicationDocuments = future.get().getDocuments();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        }
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            Map<List<String>, Object> internalList = new HashMap<>();
+            List<String> pks = new ArrayList<String>();
+            aux = medicationDocument.getId();
+            pks.add(pkToDate(aux));
+            pks.add(pkToName(aux));
+            internalList.put(Collections.singletonList("dateName"), pks);
+            internalList.put(Collections.singletonList("body"),
+                    medicationDocument.toObject(MedicationEntity.class));
+            externalList.add(internalList);
+        }
+        return externalList;
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationsBetween(String owner, String petName,
+                                                                    String initialDate, String finalDate) {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        String aux;
+        ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments = null;
+        try {
+            medicationDocuments = future.get().getDocuments();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        }
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            Map<List<String>, Object> internalList = new HashMap<>();
+            List<String> pks = new ArrayList<String>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Explicit type arguments can be replaced by Diamond Operator](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)